Classes/Report/SiteHandlingStatus.php

```diff
@@ -0,0 +1,148 @@
+
+use ApacheSolrForTypo3\Solr\Domain\Site\SiteRepository;
+use ApacheSolrForTypo3\Solr\Domain\Site\Typo3ManagedSite;
+use ApacheSolrForTypo3\Solr\System\Configuration\ExtensionConfiguration;
+use Exception;
+use TYPO3\CMS\Core\Site\Entity\Site as Typo3Site;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Reports\Status;
+
+/**
+ * Provides an status report about current state of site handling configurations.
+ *
+ * Following thigs are checked currently:
+ * * Entry Point[base] scheme expects -> http[s]
+ * * Entry Point[base] authority expects -> [user-info@]host[:port]
+ */
+class SiteHandlingStatus extends AbstractSolrStatus
+{
+    const TITLE_SITE_HANDLING_CONFIGURATION = 'Site handling configuration';
+
+    /**
+     * Site Repository
+     *
+     * @var SiteRepository
+     */
+    protected $siteRepository = null;
+
+    /**
+     * @var ExtensionConfiguration
+     */
+    protected $extensionConfiguration = null;
+
+    /**
+     * SolrStatus constructor.
+     * @param ExtensionConfiguration $extensionConfiguration
+     * @param SiteRepository|null $siteRepository
+     */
+    public function __construct(ExtensionConfiguration $extensionConfiguration = null, SiteRepository $siteRepository = null)
+    {
+        $this->extensionConfiguration = $extensionConfiguration ?? GeneralUtility::makeInstance(ExtensionConfiguration::class);
+        $this->siteRepository = $siteRepository ?? GeneralUtility::makeInstance(SiteRepository::class);
+    }
+
+    /**
+     * @return array
+     * @throws Exception
+     */
+    public function getStatus()
+    {
+        $reports = [];
+
+        if ($this->extensionConfiguration->getIsAllowLegacySiteModeEnabled()) {
+            $reports[] = GeneralUtility::makeInstance(
+                Status::class,
+                /** @scrutinizer ignore-type */ self::TITLE_SITE_HANDLING_CONFIGURATION,
+                /** @scrutinizer ignore-type */ 'The lagacy site mode is enabled.',
+                /** @scrutinizer ignore-type */ 'The legacy site mode is not recommended and will be removed in EXT:Solr 11. Please switch to site handling as soon as possible.',
+                /** @scrutinizer ignore-type */ Status::WARNING
+            );
+            return $reports;
+        }
+
+        foreach ($this->siteRepository->getAvailableSites() as $site) {
+            if (!($site instanceof Typo3ManagedSite)) {
+                $reports[] = GeneralUtility::makeInstance(
+                    Status::class,
+                    /** @scrutinizer ignore-type */ self::TITLE_SITE_HANDLING_CONFIGURATION,
+                    /** @scrutinizer ignore-type */ 'Something went wrong',
+                    /** @scrutinizer ignore-type */ vsprintf('The configured Site "%s" is not TYPO3 managed site. Please refer to TYPO3 site management docs and configure the site properly.', [$site->getLabel()]),
+                    /** @scrutinizer ignore-type */ Status::ERROR
```

Comment:
Shouldn't we just show a warning here since using the lecay mode is not recommended but still supported?